ExampleApp/ExampleApp/Common/Services/RxBluetoothKitService.swift

```diff
@@ -0,0 +1,137 @@
+
+    // MARK: - Private fields
+    private let centralManager: CentralManager = CentralManager(queue: .main)
+
+    private let scheduler: ConcurrentDispatchQueueScheduler
+
+    private let disposeBag = DisposeBag()
+
+    private var connectedPeripherals: [Peripheral] = []
+
+    private var scanningDisposable: Disposable!
+
+    // MARK: - Initialization
+    init() {
+        let timerQueue = DispatchQueue(label: Constant.Strings.defaultDispatchQueueLabel)
+        scheduler = ConcurrentDispatchQueueScheduler(queue: timerQueue)
+    }
+
+
+    // MARK: - Scanning for peripherals
+
+    // You start from observing state of your CentralManager object. Within RxBluetoothKit v.5.0, it is crucial
+    // that you use .startWith(:_) operator, and pass the initial state of your CentralManager with
+    // centralManager.state.
+    func startScanning() {
+        scanningDisposable = centralManager.observeState()
+                .startWith(centralManager.state)
+                .filter {
+                    $0 == .poweredOn
+                }
+                .subscribeOn(MainScheduler.instance)
+                .timeout(4.0, scheduler: scheduler)
+                .flatMap { [unowned self] _ -> Observable<ScannedPeripheral> in
+                    return self.centralManager.scanForPeripherals(withServices: nil)
+                }.bind(to: scanningSubject)
+    }
+    // If you wish to stop scanning for peripherals, you need to dispose the Disposable object, that is created when
+    // you either subscribe for events from an observable returned by centralManager.scanForPeripherals(:_), or you bind
+    // it to an observer. Check starScanning() above for details.
+    func stopScanning() {
+        scanningDisposable.dispose()
+    }
+
+
+    // MARK: - Discovering Services
+
+    // When you discover a service, first you need to establish a connection with a peripheral. Then you call
+    // discoverServices(:_) that peripheral object.
+    func discoverServices(for peripheral: Peripheral) {
+        centralManager.establishConnection(peripheral)
+                .do(onNext: { [unowned self] _ in
+                    self.addConnected(peripheral)
+                    self.observeDisconnect(for: peripheral)
+                })
+                .flatMap {
+                    $0.discoverServices(nil)
+                }.bind(to: servicesSubject)
+                .disposed(by: disposeBag)
+    }
+
+
+    // MARK: - Discovering Characteristics
+    func discoverCharacteristics(for service: Service) -> Observable<[Characteristic]> {
+        return service.discoverCharacteristics(nil).asObservable()
+    }
+
+    // MARK: - Utility functions
+
+    // You might be connected with more than one peripheral at a time, so it's a good decision to keep a collection
+    // of currently connected peripherals.
+    private func addConnected(_ peripheral: Peripheral) {
+        let peripherals = connectedPeripherals.filter {
```

Comment:
You can use `Set` instead an array. It will provide there is no duplicate in a Container.

Reply:
It's impossible atm, because Peripheral doesn't conform to Hashable.